EIPS/eip-7577.md

```diff
@@ -0,0 +1,107 @@
+author: danceratopz (@danceratopz), Ahmad Bitar (@smartprogrammer93)
+discussions-to: https://ethereum-magicians.org/t/add-eip-versioning-scheme-for-eips/17295
+status: Draft
+type: Meta
+created: 2023-12-13
+---
+
+## Abstract
+
+This EIP introduces a versioning scheme for [Standards Track](./eip-1.md#eip-types) EIPs by applying semantic versioning based on changes made to the EIP's Specification section once its status has changed from `Draft` to `Review`.
+
+## Motivation
+
+EIP specifications often receive increasing modifications as more people review them, which is generally the case as client teams start implementing the specifications and the community gains a better understanding of their interaction with the rest of the protocol. These changes can be difficult to track. In particular, as EVM reference tests are often not maintained (and generally not released) by client teams or the EIP's authors, it can be difficult to ascertain whether a release of reference tests (for example from ethereum/tests or ethereum/execution-spec-tests) is sufficient, or even valid, to test the latest version of an EIP's specifications or the specification as currently implemented by a client.
+
+This EIP proposes a semantic versioning scheme and an addition of a CHANGELOG section for EIPs that allows the community to easily track changes, their impact on current client and test implementations and maintain a clear history.
+
+## Specification
+
+Once an EIP has moved out of "Draft" status, it MUST use the EIP versioning scheme outlined below. It MAY already use the versioning scheme in "Draft" status, which could be useful if the specification is actively being implemented. If more than one team is implementing the specification, it is RECOMMENDED to change the EIP's status to "Review".
+
+The EIP versioning scheme MUST follow a semantic versioning scheme of `MAJOR.MINOR.PATCH`, which is applied as follows:
+
+1. `MAJOR`: A breaking change to the specifications that requires an implementation change and a change to the reference tests.
+2. `MINOR`: An addition to the specifications that does not require changing an existing implementation, but requires additional implementation and additional test coverage to be added to the reference tests.
+3. `PATCH`: Any cosmetic change to, or a reformulation of, the EIP without specification change.
+
+Before the EIP has moved out of Draft status and is being versioned, the version number MUST initially have `MAJOR` version `0`.
+
+For every change made to an EIP via a PR made to ethereum/EIPs, the PR author MUST add a new entry to the CHANGELOG section of the EIP that outlines the changes made within the PR. This CHANGELOG entry MUST include the following:
+
+1. A new version number that follows the semantic versioning scheme outlined above.
+2. The date when the changes where introduced.
+3. A link to the ethereum/EIPs PR that implements the changes.
```

Comment:
The current rules forbid links to PRs. This isn't a formatting issue, so you'll have to talk to EIP editors if you want this change.

Reply:
Removed link requirement in 2a0eb65dbddbeb299ad419b0cfc8fcef82dcc669.

I think a link to the changes would be helpful to readers - I'll follow up with EIP editors to gauge interest.